src/redteam/commands/discover.ts

```diff
@@ -0,0 +1,285 @@
+import chalk from 'chalk';
+import cliProgress from 'cli-progress';
+import { type Command, Option } from 'commander';
+import * as fs from 'fs';
+import yaml from 'js-yaml';
+import path from 'path';
+import { z } from 'zod';
+import cliState from '../../cliState';
+import { VERSION } from '../../constants';
+import { getEnvString } from '../../envars';
+import { getUserEmail } from '../../globalConfig/accounts';
+import { cloudConfig } from '../../globalConfig/cloud';
+import logger, { setLogLevel } from '../../logger';
+import { loadApiProvider, loadApiProviders } from '../../providers';
+import telemetry from '../../telemetry';
+import type { ApiProvider, UnifiedConfig } from '../../types';
+import { setupEnv } from '../../util';
+import { getProviderFromCloud } from '../../util/cloud';
+import { readConfig } from '../../util/config/load';
+import { writePromptfooConfig } from '../../util/config/manage';
+import invariant from '../../util/invariant';
+import { getRemoteGenerationUrl } from '../remoteGeneration';
+
+const ArgsSchema = z.object({
+  config: z.string().optional(),
+  output: z.string().optional(),
+  target: z.string().optional(),
+  envPath: z.string().optional(),
+  verbose: z.boolean().optional(),
+  preview: z.boolean().optional(),
+  turns: z.number().optional(),
+});
+
+type Args = z.infer<typeof ArgsSchema>;
+/**
+ * Queries Cloud for the purpose-discovery logic, sends each logic to the target,
+ * and summarizes the results.
+ *
+ * @param target - The target API provider.
+ * @param maxTurns - The maximum number of turns to run the discovery process.
+ * @returns The purpose of the target.
+ */
+export async function doTargetPurposeDiscovery(
+  target: ApiProvider,
+  maxTurns?: number,
+): Promise<string> {
+  const conversationHistory: { type: 'promptfoo' | 'target'; content: string }[] = [];
+
+  let turnCounter = 1;
+
+  const pbar = new cliProgress.SingleBar({
+    format: `Discovering purpose {bar} {percentage}% | {value}/${maxTurns ? '{total}' : '∞'} turns`,
```

Comment:
Would the infinity freak people out / cause confusion? Maybe omit the fraction part if maxTurns isn't supplied.